lib/collection/src/shards/local_shard/mod.rs

```diff
@@ -969,15 +970,48 @@ impl LocalShard {
         vector_size * info.points_count
     }
 
-    pub async fn local_shard_info(&self) -> CollectionInfoInternal {
+    pub async fn local_shard_status(&self) -> (ShardStatus, OptimizersStatus) {
+        let mut status = ShardStatus::Green;
+        let mut optimizer_status = OptimizersStatus::Ok;
+
+        {
+            let segments = self.segments().read();
+            for (_, segment) in segments.iter() {
+                let segment_info = segment.get().read().info();
+                if segment_info.segment_type == SegmentType::Special {
+                    status = ShardStatus::Yellow;
+                }
+
+                if !segments.failed_operation.is_empty() || segments.optimizer_errors.is_some() {
+                    status = ShardStatus::Red;
+                }
```

Comment:
We probably need to break here to not overwrite a red status with something else after.

```suggestion
                if !segments.failed_operation.is_empty() || segments.optimizer_errors.is_some() {
                    status = ShardStatus::Red;
                    break;
                }
```

Reply:
Looks like segment is a list. I did some refactoring